preludezdev/NaverMovieSearchApi2/app/src/main/java/com/example/navermoviesearchapi2/presentation/main/MainActivity.kt

```diff
@@ -0,0 +1,154 @@
+package com.example.navermoviesearchapi2.presentation.main
+
+import android.app.Activity
+import android.content.Intent
+import android.os.Bundle
+import android.text.Editable
+import android.text.TextWatcher
+import android.view.KeyEvent
+import android.view.inputmethod.InputMethodManager
+import androidx.lifecycle.Observer
+import androidx.recyclerview.widget.DividerItemDecoration
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.example.navermoviesearchapi2.R
+import com.example.navermoviesearchapi2.base.BaseActivity
+import com.example.navermoviesearchapi2.databinding.ActivityMainBinding
+import com.example.navermoviesearchapi2.presentation.detail.DetailActivity
+import com.example.navermoviesearchapi2.utils.plusAssign
+import io.reactivex.BackpressureStrategy
+import io.reactivex.Observable
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.subjects.BehaviorSubject
+import io.reactivex.subjects.Subject
+import kotlinx.android.synthetic.main.activity_main.*
+import org.koin.androidx.viewmodel.ext.android.viewModel
+import java.util.concurrent.TimeUnit
+
+class MainActivity : BaseActivity<ActivityMainBinding>(R.layout.activity_main) {
+
+    private val compositeDisposable = CompositeDisposable()
+    private val backButtonSubject: Subject<Long> = BehaviorSubject.createDefault(0L).toSerialized()
```

Comment:
```suggestion
    private val backButtonSubject = BehaviorSubject.createDefault(0L).toSerialized()
```
타입을 생략할 수도 있지않을까 싶습니다

Reply:
수정했습니다
ee65f9b